src/Operation/Factory/SubresourceOperationFactory.php

```diff
@@ -91,11 +91,14 @@ private function computeSubresourceOperations(string $resourceClass, array &$tre
             // Handle maxDepth
             if ($rootResourceClass === $resourceClass) {
                 $maxDepth = $subresource->getMaxDepth();
+                // When we return to rootResourceClass for another subresource property, reset the depth
+                $depth = 0;
```

Comment:
shouldn't this value be updated after the test at line 98?

Reply:
For circular dependency for example? If you return to the rootRecourceClass by circular dep it could reset this to 0 and go forward. Seems you right about this. I will test after line 98 and see what it does.